app/javascript/src/component_expander.js

```diff
@@ -219,11 +265,11 @@ class Expander {
   }
 
   #isValidActivatorElement(source) {
-    return source.is('button') || source.is('input[type="checkbox"]') || source.is('input[type="radio"]')
+    return source.is('button') || source.is('summary') || this.#isInput(source);
   }
 
   #isInput($activator) {
-     return $activator.is('input[type="checkbox"]') || $activator.is('input[type="radio"]')
+   return $activator.is('input[type="checkbox"]') || $activator.is('input[type="radio"]')
   }
 
 }
```

Comment:
Initially I was going to ask why we were adding in a bunch of stuff to specifically handle `details/summary` when that does it natively, but I think I understand now:

 - We use details/summary because 'design system'
 - We enhance it with an `Expander` component because we want to be able to control the `details` to expand when we check the checkbox

Correct? 

In which case I think this is OK. The only downside I can see is that there is no indication for screenreader users on the checkbox that it will reveal more content - but I think that's solvable... I'll add it in another comment

Reply:
You are correct.

Without augmenting the native browser functionality of details/summary to, at least, track the internal openState, I found the Checkbox could get confused as to whether it was open, or not. 

I didn't add any 'aria-controls' (for example) to the Checkbox because it doesn't really control it, at least in the proper sense. If you're using a screenreader, after checking that box to say you want analytics, by the time you reach the details/summary, I am assuming the details/summary will report its state as open (or whatever) and work as though it was already in an open state, etc. The checkbox is not an open/close control, as such, but more a  convenience feature that means they do not need to perform an extra click by the time they read it. If they have not clicked the checkbox and go straight to the details/summary, presumably they would interact with it in whatever normal fashion.

The intended use was to make sure (and I guess it is the same for all users) that, once the checkbox is ticked, users do not miss the (hidden but also required) configuration options.